examples/preact-spa/src/register.tsx

```diff
@@ -1,21 +1,65 @@
 import { RegistrationFlow, UpdateRegistrationFlowBody } from "@ory/client"
 import { UserAuthCard } from "@ory/elements-preact"
-import { useEffect, useState } from "preact/hooks"
-import sdk from "./sdk"
+import { useCallback, useEffect, useState } from "preact/hooks"
+import { getSearchParam, sdk, sdkError } from "./sdk"
 import { useLocation } from "wouter"
 
 export const Register = () => {
   const [flow, setFlow] = useState<RegistrationFlow | null>(null)
 
   const [location, setLocation] = useLocation()
 
-  useEffect(() => {
+  // Get the flow based on the flowId in the URL (.e.g redirect to this page after flow initialized)
+  const getFlow = useCallback(
+    (flowId: string) =>
+      sdk
+        // the flow data contains the form fields, error messages and csrf token
+        .getRegistrationFlow({ id: flowId })
+        .then(({ data: flow }) => setFlow(flow))
+        .catch(sdkErrorHandler),
+    [],
+  )
+
+  // initialize the sdkError for generic handling of errors
+  const sdkErrorHandler = sdkError(getFlow, setFlow, "/registration", true)
+
+  // create a new registration flow
+  const createFlow = () => {
     sdk
+      // we don't need to specify the return_to here since we are building an SPA. In server-side browser flows we would need to specify the return_to
       .createBrowserRegistrationFlow()
-      .then(({ data: flow }) => {
-        setFlow(flow)
+      .then(({ data: flow }) => setFlow(flow))
+      .catch(sdkErrorHandler)
+  }
+
+  // submit the registration form data to Ory
+  const submitFlow = (body: UpdateRegistrationFlowBody) => {
+    // something unexpected went wrong and the flow was not set
+    if (!flow) return setLocation("/signup", { replace: true })
```

Comment:
```suggestion
    if (!flow) return setLocation("/registration", { replace: true })
```

this was changed to `registration`, right?

Reply:
Yes, Changed to default value that Kratos has.